Tools/cases_generator/analysis.py

```diff
@@ -414,3 +414,61 @@ def check_macro_components(
                 case _:
                     assert_never(uop)
         return components
+
+    def report_non_viable_uops(self, jsonfile: str) -> None:
+        print("The following ops are not viable uops:")
+        skips = {
```

Comment:
Thinking aloud a bit here, and this can wait for another PR.

Rhetorical question: Why are these not viable micro-ops?
Ideally, we would recognize the properties that make them not viable and only report uops that are not viable if we expect to be viable.

We want to add properties/decorators to ops/instructions for things like purity. Could we do that for this as well.
E.g.
`@unreachable` for `CACHE` and `RESERVED`, `@super` for `LOAD_FAST_LOAD_FAST`, etc. `@branch` for `POP_JUMP_IF_TRUE` and `POP_JUMP_IF_FALSE`, and so on...

Reply:
Since @Fidget-Spinner is also interested in  this, I'll wait for him to make a specific proposal. Likely the syntax will be `pure op(X, (...)) { ... }` and so on.